app/models/article.js

```diff
@@ -40,5 +44,22 @@ function categoryValidation(val) {
     return CATEGORIES_LIST.find(category => category.key === val)
 }
 
+async function getLink(link, docType, docId, n = 1) {
+    const newLink = n === 1 ? link : `${link}-${n}`
+    const doc = await Link.findOne({link: newLink}).exec()
+    if(!doc) {
+        Link.create({link: newLink, docType, doc_id: docId})
+        return link
+    }
+    return await getLink(link, docType, docId, n + 1)
+}
+
+articleSchema.pre('validate', async function(next) {
+    if(!this.isModified('title')) return next()
+    const sluggedLink = slugify(this.title, SLUGIFY_OPTIONS)
+    this.link = await getLink(sluggedLink, ARTICLE_TYPE, this._id)
```

Comment:
can `getnLink` crash? should we try...catch it?

Reply:
Should we create a link during validation? Validation sounds like a thing that shouldn't modify the data inputted inside.

Reply:
This occurs pre-validation to ensure schema property rules are enforced.